app/src/main/java/com/github/mik629/android/fundamentals/ui/movieslist/FragmentMoviesList.kt

```diff
@@ -1,72 +1,75 @@
-import com.github.mik629.android.fundamentals.data.mappers.ActorMapper
-import com.github.mik629.android.fundamentals.data.network.ServerApi
-import com.github.mik629.android.fundamentals.data.repositories.MoviesRepositoryImpl
 import com.github.mik629.android.fundamentals.databinding.FragmentMoviesListBinding
 import com.github.mik629.android.fundamentals.di.AppModule
+import com.github.mik629.android.fundamentals.di.buildGlideRequest
 import com.github.mik629.android.fundamentals.ui.global.MovieItemAdapter
 import com.github.mik629.android.fundamentals.ui.moviedetails.FragmentMovieDetails
-import com.github.mik629.android.fundamentals.vm.MoviesListViewModel
+import com.google.android.material.snackbar.BaseTransientBottomBar.LENGTH_LONG
+import com.google.android.material.snackbar.Snackbar
 
-class FragmentMoviesList : Fragment() {
-    private lateinit var binding: FragmentMoviesListBinding
+class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
+    private val binding by viewBinding(FragmentMoviesListBinding::bind)
 
-    private val viewModel by lazy {
-        MoviesListViewModel(
-            MoviesRepositoryImpl(
-                AppModule().retrofit.create(ServerApi::class.java),
-                ActorMapper()
-            )
-        )
-    }
+    private val viewModel: MoviesListViewModel by viewModels(
+        factoryProducer = {
+            MoviesListViewModelFactory(requireContext())
+        }
+    )
 
     private val movieItemAdapter by lazy {
         MovieItemAdapter(
-            { movieTitle ->
-                requireFragmentManager()
+            { movie ->
+                parentFragmentManager
                     .beginTransaction()
-                    .addToBackStack(FragmentMoviesList::class.simpleName)
-                    .add(R.id.main_container, FragmentMovieDetails.newInstance(movieTitle))
+                    .addToBackStack(FragmentMoviesList::class.qualifiedName) // fixme add cicerone
+                    .add(R.id.main_container, FragmentMovieDetails.newInstance(movie.id))
                     .commit()
             },
-            GlideApp.with(this)
-                .asDrawable()
-                .thumbnail(0.1f)
-                .placeholder(R.drawable.ic_image_loading)
-                .fallback(R.drawable.ic_broken_image)
-                .transition(DrawableTransitionOptions.withCrossFade())
-                .fitCenter()
+            buildGlideRequest(this)
         )
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        binding = FragmentMoviesListBinding.inflate(layoutInflater)
-
-        with(binding) {
-            movieList.adapter = movieItemAdapter
-            viewModel.movies.observe(this@FragmentMoviesList.viewLifecycleOwner) {
-                movieItemAdapter.submitList(it)
-            }
+    override fun onStart() {
```

Comment:
observe in onStart ?
try to hide app(to onStop be called)few times and calculate how many times movieItemAdapter.submitList(it) was called in onStart

Reply:
?